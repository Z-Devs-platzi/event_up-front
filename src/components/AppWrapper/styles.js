import styled from 'styled-components';

export const Wrapper = styled.div`
  display: grid;
  width: 100vw;
  height: 100vh;
  grid-template-rows: 90px auto;
  grid-template-columns: 248px auto;
  grid-template-areas:
    'nav header'
    'nav content';
  @media screen and (max-width: 900px) {
    grid-template-rows: 56px 90px auto 60px;
    grid-template-columns: auto;
    grid-template-areas:
      'nav'
      'header'
      'content'
      'foot';
  }
`;
export const WrapperContent = styled.div`
  grid-area: content;
  padding: 5px;
`;
export const WrapperHeader = styled.div`
  grid-area: header;
`;
export const NavContainer = styled.div`
  grid-area: nav;
`;
