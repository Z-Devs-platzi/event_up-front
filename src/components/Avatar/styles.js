import styled from 'styled-components';

export const AvatarImage = styled.img`
  --size: ${(props) => props.size};
  display: block;
  width: var(--size);
  height: var(--size);
  background-color: var(--grey);
  border-radius: 50%;
  object-fit: cover;
  border: none;
`;
