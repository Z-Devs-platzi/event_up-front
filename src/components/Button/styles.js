import styled from 'styled-components';

export const Buttons = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 1em;
  padding: 0.5rem 2rem;
  background: ${(props) =>
    props.buttonType === 'primary' ? 'var(--secondary-color)' : 'transparent'};
  border: ${(props) =>
    props.buttonType === 'primary'
      ? '1px solid var(--secondary-color)'
      : '1px solid var(--black)'};
  color: var(--black);
  cursor: pointer;
  font-size: 1rem;
  font-weight: 700;
  border-radius: 0.5rem;
  transition: background 0.3s ease-in-out;
  &:hover {
    background: ${(props) =>
      props.buttonType === 'primary'
        ? 'var(--white)'
        : 'var(--secondary-color)'};
  }
`;
