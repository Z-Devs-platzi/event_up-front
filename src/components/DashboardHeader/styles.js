import styled from 'styled-components';
import { BiChevronDown } from 'react-icons/bi';

export const HeaderWrapper = styled.header`
  display: flex;
  align-items: center;
  padding: 1.25rem 1rem;
  background-color: var(--light-green);
`;

export const HeaderTitle = styled.h3`
  display: flex;
  align-items: center;
  margin-left: 1rem;
  font-size: 1.25rem;
  cursor: pointer;
`;
