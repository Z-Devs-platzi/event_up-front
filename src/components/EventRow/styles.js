import styled from 'styled-components';

export const Row = styled.div`
  display: grid;
  grid-template-columns: 44px 3fr 1fr 90px 80px;
  grid-gap: 20px;
  place-items: center;
  height: 92px;
  padding: 0 20px;
  font-weight: 600;
  background-color: var(--light-grey-1);
  border-radius: 8px;

  & .Row__avatar {
    grid-column: 1 / span 1;
  }

  & .Row__title {
    grid-column: 2 / span 1;
    justify-self: start;

    & .name {
      margin-bottom: 8px;
    }

    & .update-time {
      font-size: 0.75rem;
      color: var(--grey);
    }
  }

  & .Row__date {
    grid-column: 3 / span 1;
    justify-self: start;

    & .date {
      margin-bottom: 8px;
    }

    & .time {
      font-size: 0.75rem;
      color: var(--grey);
    }
  }

  & .Row__status {
    grid-column: 4 / span 1;
    justify-self: end;
  }

  & .Row__options {
    grid-column: 5 / span 1;
    display: flex;
    flex-wrap: nowrap;
    align-items: center;

    li {
      padding: 0 1px;

      & .action {
        display: block;
        font-size: 0;
        color: var(--black);

        &:hover {
          color: var(--secondary-color);
        }

        & .icon {
          width: 24px;
          height: 24px;
        }
      }
    }
  }

  @media screen and (max-width: 570px) {
    grid-template-columns: 44px 1fr 80px;
    grid-template-rows: 24px 1fr 22px 24px;
    grid-gap: 10px;
    place-items: initial;
    height: initial;
    margin-bottom: 20px;
    padding: 10px 10px 15px 15px;

    & .Row__avatar {
      grid-column: 1 / span 1;
      grid-row: 2 / span 1;
    }

    & .Row__title {
      grid-column: 2 / span 2;
      grid-row: 2 / span 1;
      width: 100%;
      padding-bottom: 5px;
      border-bottom: 1px solid var(--light-grey-2);
    }

    & .Row__date {
      grid-column: 2 / span 2;
      grid-row: 3 / span 1;

      & .date {
        display: inline-block;
        width: 130px;
        margin-bottom: 0;
      }

      & .time {
        display: inline;
      }
    }

    & .Row__status {
      grid-column: 2 / span 2;
      grid-row: 4 / span 1;
      justify-self: start;
    }

    & .Row__options {
      grid-column: 3 / span 1;
      grid-row: 1 / span 1;
    }
  }
`;
