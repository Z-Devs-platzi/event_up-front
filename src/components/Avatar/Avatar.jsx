import React from 'react';
import PropTypes from 'prop-types';
import { AvatarImage } from './Avatar.styles';

const Avatar = ({ size, src, alt }) => (
  <AvatarImage size={size} src={src} alt={alt} />
);

Avatar.propTypes = {
  size: PropTypes.string,
  src: PropTypes.string,
  alt: PropTypes.string,
};

Avatar.defaultProps = {
  size: '2.75rem',
  src:
    'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=80',
  alt: 'Avatar',
};

export default Avatar;
