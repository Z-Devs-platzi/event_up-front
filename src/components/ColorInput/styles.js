import styled from 'styled-components';
import { CenterDiv } from '../../styles/GlobalStyles';

export const InputSwatch = styled.div`
  display: inline-flex;
  position: relative;
  align-items: center;
  justify-content: center;
  background: '#fff';
  border-radius: 50%;
  border: 1px solid #808080;
  boxshadow: var(--shadow);
  cursor: pointer;
  width: 40px;
  height: 40px;
  margin: 5px;
`;
export const InputPopOver = styled.div`
  position: absolute;
  top: 50px;
  left: 160px;
  z-index: 2;
`;
export const InputCover = styled.div`
  position: fixed;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
`;
export const InputColor = styled.div`
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: ${(props) => props.color};
`;

export const LabelInput = styled.label`
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  color: var(--grey);
`;
export const ColorContainerInput = styled(CenterDiv)`
  width: 168px;
  height: 40px;
`;
