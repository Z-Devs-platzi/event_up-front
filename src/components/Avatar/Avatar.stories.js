import React from 'react';
import Avatar from '.';
import { GlobalStyle } from '../../styles/GlobalStyles';

const Story = (args) => {
  return (
    <>
      <GlobalStyle />
      <Avatar {...args} />
    </>
  );
};

export const AvatarDefault = Story.bind({});
AvatarDefault.args = {
  size: '2.75rem',
  src:
    'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=80',
  alt: 'Avatar',
};

export const AvatarLanding = Story.bind({});
AvatarLanding.args = {
  size: '10rem',
  src:
    'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=80',
  alt: 'Speaker Name',
};

export default {
  title: 'Atom/Avatar',
  component: Avatar,

  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/,
};
