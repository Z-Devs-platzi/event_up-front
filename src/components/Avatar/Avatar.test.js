import React from 'react';
import { mount } from 'enzyme';
import Avatar from '.';
import { AvatarImage } from './styles';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

// Component
describe('<Avatar />', () => {
  test('Avatar / Default', () => {
    const defaultAvatar = mount(<Avatar />);
    expect(defaultAvatar.length).toEqual(1);
  });
});

// Styles
describe('Avatar / Styles', () => {
  it('Avatar Styles / Default', () => {
    const avatarDefault = renderer.create(<Avatar />).toJSON();
    expect(avatarDefault).toMatchSnapshot();
    expect(avatarDefault).toHaveStyleRule('width', 'var(--size)');
  });
});
