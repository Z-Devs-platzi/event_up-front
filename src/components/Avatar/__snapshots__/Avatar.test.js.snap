// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar / Styles Avatar Styles / Default 1`] = `
.c0 {
  --size: 2.75rem;
  display: block;
  width: var(--size);
  height: var(--size);
  background-color: var(--grey);
  border-radius: 50%;
  object-fit: cover;
  border: none;
}

<img
  alt="Avatar"
  className="c0"
  size="2.75rem"
  src="https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=80"
/>
`;
